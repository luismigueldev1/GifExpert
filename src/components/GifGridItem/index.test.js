import React from "react";
import { shallow } from "enzyme";
import GifGridItem from ".";

describe("Pruebas para GifGridItem", () => {
  const title = "yu gi oh GIF";
  const url =
    "https://media2.giphy.com/media/11zNrvy5Xh5gFa/giphy.gif?cid=cf8f0d8045ww5upcupkqnscdzxh72meo24oob66x9iltsqh5&rid=giphy.gif&ct=g";

  const wrapper = shallow(<GifGridItem title={title} url={url} />);

  test("debe retornar el componente y hacer match con el snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("debe tener un <p> con un title", () => {
    const p = wrapper.find("p");
    expect(p.text().trim()).toBe(title);
  });

  test("debe tener un <img> igual al url y alt de props", () => {
    const img = wrapper.find("img");
    //img.prop("alt")
    //console.log(img.props())
    expect(img.prop("src")).toBe(url);
    expect(img.prop("alt")).toBe(title);
  });

  test("debe tener la clase animate__fadeIn", () => {
    const div = wrapper.find("div");
    // const className = div.prop("className");
    // expect(className.includes("animate__fadeIn")).toBe(true);
    expect(div.hasClass("animate__fadeIn")).toEqual(true);
  });
});
