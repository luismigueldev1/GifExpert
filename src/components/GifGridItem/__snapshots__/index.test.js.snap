// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas para GifGridItem debe retornar el componente y hacer match con el snapshot 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="yu gi oh GIF"
    src="https://media2.giphy.com/media/11zNrvy5Xh5gFa/giphy.gif?cid=cf8f0d8045ww5upcupkqnscdzxh72meo24oob66x9iltsqh5&rid=giphy.gif&ct=g"
  />
  <p>
    yu gi oh GIF
  </p>
</div>
`;
